### PR TITLE
Add port to the Host header if port is set

### PR DESCRIPTION
**Pull request by jonnymacs**
https://tools.ietf.org/html/rfc2616#section-14.23

Use path as request-uri if host Header is set
https://tools.ietf.org/html/rfc2616#section-5.1.2

### CODE REVIEW

File: Sources/Prorsum/HTTP/Serializer/RequestSerializer.swift [hunk 1]
```diff
@@ -19,7 +19,23 @@ public struct RequestSerializer {
         let newLine: [UInt8] = [13, 10]
         
         let method = "\(request.method)".uppercased()
-        try stream.write("\(method) \(request.url.absoluteString) HTTP/\(request.version.major).\(request.version.minor)".bytes, deadline: deadline)
+
+        // if host is set
+        // build uri from path & query
+        // else use absoluteString
+        //
+        let request_uri: String
```
Comment (noppoMan) [inline Sources/Prorsum/HTTP/Serializer/RequestSerializer.swift]: Variable name should be camel case.
See https://swift.org/documentation/api-design-guidelines/

File: Sources/Prorsum/HTTP/HTTPClient.swift [hunk 2]
```diff
@@ -96,7 +96,11 @@ public class HTTPClient {
         }
         
         if request.host == nil {
-            request.host = url.host
+            if(url.port != nil) {
```
Comment (noppoMan) [inline Sources/Prorsum/HTTP/HTTPClient.swift]: I think

```swift
if let host = request.host, let port = url.port {
    request.host = "\(host):\(port)"
} else {
```

is more safety and swifty

File: Sources/Prorsum/HTTP/Serializer/RequestSerializer.swift [hunk 3]
```diff
@@ -19,7 +19,23 @@ public struct RequestSerializer {
...
+        if request.host != nil {
+            var uri = !request.url.path.isEmpty ? request.url.path : "/" 
+            if let query = request.url.query {
+              uri += "?\(query)"
```
Comment (noppoMan) [inline Sources/Prorsum/HTTP/Serializer/RequestSerializer.swift]: indent seems 2 space

**Review by jonnymacs — COMMENTED** [review]
@noppoMan changes as requested 👍

**Comment by noppoMan** [comment]
Thanks👍 I’ll merge this when I get home.

**Comment by noppoMan** [comment]
I published version 0.3.3 that include your changes.

**Pull request merged**